File: prayer_app_expo/src/components/PrayerRow.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { SalahLayout, interFont, getPrayerIcon } from '../theme/theme';
import { useTheme } from '../providers/ThemeProvider';

export default function PrayerRow({ name, time, isHighlighted = false }) {
    const { theme: tc } = useTheme();
    const textColor = isHighlighted ? tc.accent : tc.textPrimary;
    const iconColor = isHighlighted ? tc.accent : tc.textMuted;
    const iconName = getPrayerIcon(name);

    const borderColor = isHighlighted
        ? hexToRgba(tc.accent, SalahLayout.rowBorderOpacity)
        : 'transparent';

    return (
        <View style={[
            styles.row,
            isHighlighted && {
                backgroundColor: tc.card,
                borderWidth: SalahLayout.rowBorderWidth,
                borderColor: borderColor,
            },
        ]}>
            <MaterialCommunityIcons
                name={iconName}
                size={SalahLayout.rowIconSize}
                color={iconColor}
            />
            <Text style={[styles.name, { color: textColor }]}>{name}</Text>
            <View style={styles.spacer} />
            <Text style={[styles.time, { color: textColor }]}>{time}</Text>
        </View>
    );
}

function hexToRgba(hex, alpha) {
    const h = hex.replace('#', '');
    const r = parseInt(h.substring(0, 2), 16);
    const g = parseInt(h.substring(2, 4), 16);
    const b = parseInt(h.substring(4, 6), 16);
    return `rgba(${r}, ${g}, ${b}, ${alpha})`;
}

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row',
        alignItems: 'center',
        height: SalahLayout.rowHeight,
        paddingHorizontal: SalahLayout.rowPaddingH,
        borderRadius: SalahLayout.rowRadius,
    },
    name: {
        fontFamily: interFont('500'),
        fontSize: SalahLayout.rowTextSize,
        marginLeft: SalahLayout.rowPaddingH,
    },
    spacer: {
        flex: 1,
    },
    time: {
        fontFamily: interFont('500'),
        fontSize: SalahLayout.rowTextSize,
    },
});

File: prayer_app_flutter/lib/src/components/prayer_row.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';

class PrayerRow extends StatelessWidget {
  final String name;
  final String time;
  final IconData icon;
  final bool isHighlighted;

  const PrayerRow({
    super.key,
    required this.name,
    required this.time,
    required this.icon,
    this.isHighlighted = false,
  });

  @override
  Widget build(BuildContext context) {
    final textColor = isHighlighted ? AppColors.accentGold : AppColors.textPrimary;
    final iconColor = isHighlighted ? AppColors.accentGold : AppColors.textMuted;

    return Container(
      height: SalahLayout.rowHeight,
      padding: const EdgeInsets.symmetric(horizontal: SalahLayout.rowPaddingH),
      decoration: BoxDecoration(
        color: isHighlighted ? AppColors.card : Colors.transparent,
        borderRadius: BorderRadius.circular(SalahLayout.rowRadius),
        border: isHighlighted
            ? Border.all(
                color: AppColors.accentGold.withValues(alpha: SalahLayout.rowBorderOpacity),
                width: SalahLayout.rowBorderWidth,
              )
            : null,
      ),
      child: Row(
        children: [
          Icon(icon, color: iconColor, size: SalahLayout.rowIconSize),
          const SizedBox(width: SalahLayout.rowPaddingH),
          Text(
            name,
            style: AppTypography.body.copyWith(
              fontSize: SalahLayout.rowTextSize,
              fontWeight: FontWeight.w500,
              color: textColor,
            ),
          ),
          const Spacer(),
          Text(
            time,
            style: AppTypography.body.copyWith(
              fontSize: SalahLayout.rowTextSize,
              fontWeight: FontWeight.w500,
              color: textColor,
            ),
          ),
        ],
      ),
    );
  }
}
